test/chart-template.js

```diff
@@ -0,0 +1,392 @@
+        global.document.body.appendChild(container);
+        return container;
+    }
+
+    /**
+     * Creates a deep copy of entries and properties.
+     *
+     * @param {array|object} source
+     * The source to copy from.
+     *
+     * @param {object} propertiesTree
+     * The properties tree to copy.
+     *
+     * @return {array|object}
+     * The copy of the source.
+     */
+    function treeCopy(source, propertiesTree) {
+        if (!source) {
+            return source;
+        }
+        switch (typeof source) {
+            default:
+                return source;
+            case 'array':
+                return JSON.parse(JSON.stringify(source));
+            case 'object':
+                var copy = {};
+                for (var key in propertiesTree) {
+                    if (propertiesTree.hasOwnProperty(key) &&
+                        source.hasOwnProperty(key)
+                    ) {
+                        copy[key] = treeCopy(source[key], propertiesTree[key]);
+                    } else {
+                        copy[key] = undefined; // eslint-disable-line no-undefined
+                    }
+                }
+                return copy;
+        }
+    }
+    global.treeCopy = treeCopy;
+
+    /**
+     * Attachs to the update event of a chart.
+     *
+     * @param {Highcharts.Chart} chart
+     * The instance of the chart
+     *
+     * @param {Array<object>} optionsStorage
+     * The array with previous chart options
+     *
+     * @return {function}
+     * Remove function to stop the attachment
+     */
+    function attachUpdate(chart, optionsStorage) {
+        return Highcharts.addEvent(chart, 'update', function (newOptions) {
+            optionsStorage.push(treeCopy(chart.options, newOptions));
+        });
+    }
+
+    /* *
+     *
+     *  Constructor
+     *
+     * */
+
+    /**
+     * This class creates a new template for testing on generic charts. It also
+     * provides static functions to use registered templates in test cases.
+     *
+     * @param {string} name
+     * The reference name of the chart
+     *
+     * @param {Highcharts.Chart} chartConstructor
+     * The chart constructor function for the template
+     *
+     * @param {object} chartOptions
+     * The default chart Options for the template
+     *
+     * @return {ChartTemplate}
+     * The new chart template
+     */
+    function ChartTemplate(name, chartConstructor, chartOptions) {
+
+        if (!(this instanceof ChartTemplate)) {
+            return new ChartTemplate(name, chartConstructor, chartOptions);
+        }
+
+        var chart = chartConstructor(createContainer(), chartOptions),
+            testCases = [];
+
+        /* *
+         *
+         *  Instance Properties
+         *
+         * */
+
+        /**
+         * The name of the template as a flag in the chart
+         *
+         * @type {string}
+         */
+        Object.defineProperty(chart, 'template', {
```

Comment:
Will we run tests in IE < 9 ? `defineProperty` is supported since ~IE9 (https://caniuse.com/#search=defineProperty)

(Below we have more of them)

Reply:
The class does not depend on it, so I could change it for now.